app/src/main/java/com/taylorgirard/comicconvo/adapters/MessageAdapter.java

```diff
@@ -0,0 +1,149 @@
+    }
+
+    @Override
+    public int getItemCount() {
+        return messages.size();
+    }
+
+    @NonNull
+    @Override
+    public MessageViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        Context context = parent.getContext();
+        LayoutInflater inflater = LayoutInflater.from(context);
+
+        if (viewType == MESSAGE_INCOMING){
+            View contactView = inflater.inflate(R.layout.message_incoming, parent, false);
+            return new IncomingMessageViewHolder(contactView);
+        } else if (viewType == MESSAGE_OUTGOING){
+            View contactView = inflater.inflate(R.layout.message_outgoing, parent, false);
+            return new OutgoingMessageViewHolder(contactView);
+        } else{
+            throw new IllegalArgumentException("Unknown view type");
+        }
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull MessageViewHolder holder, int position) {
+        Message message = messages.get(position);
+        holder.bindMessage(message);
+    }
+
+    @Override
+    public int getItemViewType(int position) {
+        if (isMe(position)){
+            return MESSAGE_OUTGOING;
+        } else {
+            return MESSAGE_INCOMING;
+        }
+    }
+
+    private boolean isMe(int position){
+        Message message = messages.get(position);
+        return message.getSenderId() != null && message.getSenderId().equals(user.getObjectId());
+    }
+
+    public abstract class MessageViewHolder extends RecyclerView.ViewHolder{
+
+        public MessageViewHolder(@NonNull View itemView) {
+            super(itemView);
+        }
+
+        abstract void bindMessage(Message message);
+    }
+
+    public class IncomingMessageViewHolder extends MessageViewHolder{
+
+        ImageView imageOther;
+        TextView body;
+        TextView name;
+
+        public IncomingMessageViewHolder(@NonNull View itemView) {
+            super(itemView);
+            imageOther = (ImageView) itemView.findViewById(R.id.ivProfileOther);
+            body = (TextView) itemView.findViewById(R.id.tvBody);
+            name = (TextView) itemView.findViewById(R.id.tvName);
+        }
+
+        @Override
+        void bindMessage(Message message) {
+            body.setText(message.getBody());
+            try {
+                name.setText(message.getSender().fetchIfNeeded().getUsername());
+            } catch (ParseException e) {
+                e.printStackTrace();
+            }
+            ParseFile profilePic = null;
+            try {
+                profilePic = message.getSender().fetchIfNeeded().getParseFile("profilePic");
+            } catch (ParseException e) {
+                e.printStackTrace();
+            }
+            if (profilePic != null) {
+                Glide.with(context).load(profilePic.getUrl()).transform(new CircleCrop()).into(imageOther);
+            }
+        }
+    }
+
+    public class OutgoingMessageViewHolder extends MessageViewHolder{
```

Comment:
Let's move `OutgoingMessageViewHolder` and `IncomingMessageViewHolder` into their own files